src/Effect/Monad/Reader/Transformer.agda

```diff
@@ -115,7 +116,20 @@ liftReaderT MRead = record
   ; local = λ f mx → mkReaderT (λ r₂ → local f (runReaderT mx r₂))
   } where open RawMonadReader MRead
 
-open import Data.Product using (_,_)
+open import Data.Product using (_×_; _,_)
+open import Effect.Monad.Writer.Transformer.Base
+
+liftWriterT : (MR : RawMonoid r g) →
+              RawFunctor M →
+              RawMonadReader R M →
+              RawMonadReader R (WriterT (RawMonoid.Carrier MR) M)
```

Comment:
This seems really dodgy. There is no chance in hell that Agda will be able to reconstruct
`MR` simply from a `RawMonoid.Carrier MR ~ some_type` constraint e.g. `(0, max)`, `(0, +)`
and `(1, *)` are all valid monoids for the natural numbers.

Can we get rid of the `RawMonoid r g` constraint?
Otherwise we may want to parameterise `WriterT` not by a `Set` but rather a `RawMonoid`?

Reply:
Fair point, I did the latter.